### ISSUE TITLE
6.0.5 Broke passing named refcursors

### ISSUE THREAD

**Issue by rkubes** [dbeaver/dbeaver#6074]
#### System information: 
- Operating system (distribution) and version: Ubuntu 18.04.2 LTS
- DBeaver version: 6.0.5 (error), 6.0.3 (works)
- Java version: openjdk version "11.0.3" 2019-04-16
- Additional extensions: none

#### Connection specification:
- Database name and version: PostgreSQL 11.3
- Driver name: "PostgreSQL"
- Do you use tunnels or proxies (SSH, SOCKS, etc)? No

#### Describe the problem you're observing:
I have a stored procedure where I pass in the name of the REFCURSOR to store a set of results. I later use "FETCH" to get the results by name. Up to and including version 6.0.3 I can use the script editor to test this function, see the script in code below.

On 6.0.5 it doesn't properly save the REFCURSOR name, and the test fails completely.

#### Steps to reproduce, if exist:

1. Connect to a PostgreSQL database. (I'm using version 11.3, but an earlier one will probably be the same).
2. Enter the following script:
```
CREATE OR REPLACE FUNCTION
test_refcursor
(
	param_ref REFCURSOR
)
RETURNS REFCURSOR
SECURITY DEFINER
AS $$
BEGIN
	OPEN param_ref FOR
		SELECT 1 AS col1, 2 AS col2;
	RETURN param_ref;
END;
$$ LANGUAGE plpgsql;

BEGIN;
SELECT test_refcursor('my_ref');
FETCH ALL IN "my_ref";
COMMIT;
```
3. Execute script.

##### Result
```
test_refcursor|
--------------|
my_ref        |
```
##### Result - 2
```
col1|col2|
----|----|
   1|   2|
```

#### Actual behavior (6.0.5)
```
Error occurred during SQL script execution

Reason:
SQL Error [34000]: ERROR: cursor "my_ref" does not exist
```
##### Result
```
test_refcursor|
--------------|
-- refcursor  |
```
#### Include any warning/errors/backtraces from the logs

```
2019-06-08 11:09:53.246 - null
org.jkiss.dbeaver.model.sql.DBSQLException: SQL Error [34000]: ERROR: cursor "my_ref" does not exist
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.executeStatement(JDBCStatementImpl.java:134)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.executeStatement(SQLQueryJob.java:467)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.lambda$0(SQLQueryJob.java:407)
	at org.jkiss.dbeaver.model.exec.DBExecUtils.tryExecuteRecover(DBExecUtils.java:146)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.executeSingleQuery(SQLQueryJob.java:405)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.run(SQLQueryJob.java:218)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:102)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
Caused by: org.postgresql.util.PSQLException: ERROR: cursor "my_ref" does not exist
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2440)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2183)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:308)
	at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:441)
	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:365)
	at org.postgresql.jdbc.PgStatement.executeWithFlags(PgStatement.java:307)
	at org.postgresql.jdbc.PgStatement.executeCachedSql(PgStatement.java:293)
	at org.postgresql.jdbc.PgStatement.executeWithFlags(PgStatement.java:270)
	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:266)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.execute(JDBCStatementImpl.java:338)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.executeStatement(JDBCStatementImpl.java:131)
	... 7 more
```

**Comment by serge-rider** [comment]
You need to switch to transactional mode in order to manage cursors. Auto-commit imode isn't suitable for them.

**Comment by heavenlarr** [comment]
Hi I am having this issue as well after using 6.0.5. It was previously working, can you help on this? Like steps to allow this to work again. 
Thanks

**Comment by rkubes** [comment]
Also, I have tried both transactional and auto commit modes. Same behavior - 6.0.3 works, 6.0.5 fails.

**Comment by serge-rider** [comment]
Could you try 6.1?
It works in my test environment (I used sample query posted above)

**Comment by rkubes** [comment]
Still the same behavior for me in 6.1.0. In both "auto-commit" and in "transactional/manual" mode. When I run the full script, I get the error that cursor "my_ref" does not exist.

If I step through the script line by line using the "execute statement" I again still see the root cause that on the `SELECT test_refcursor('my_ref')` line, where I expect it to return "my_ref" it instead returns "-- refcursor" as noted above. This happens on both 6.0.5.

If I use 6.0.3, there's no issue. If I run the script using psql command, there's no issue. It's only an issue in dbeaver 6.0.5+.
![Screenshot from 2019-06-12 22-23-46](https://user-images.githubusercontent.com/12772522/59401594-dadffd00-8d60-11e9-8466-cb91e195670e.png)

**Comment by rkubes** [comment]
I turned on logging on my PostgreSQL server and was able to find the "root issue" so to speak.

The behavior on 6.0.3 and prior is as follows:
1) Execute `SELECT test_refcursor('my_ref');` in DBeaver 6.0.3
2) Review logs in postgres and see the expected:
```
2019-06-13 22:00:43.614 CDT [19339] LOG:  execute <unnamed>: SELECT test_refcursor('my_ref')
```

3) Execute `FETCH ALL IN "my_ref";` in DBeaver 6.0.3
4) Review logs in postgres again and see the expected:
```
2019-06-13 22:01:25.635 CDT [19339] LOG:  execute <unnamed>: FETCH ALL IN "my_ref"
```

All good so far...

1) Install dbeaver 6.1.0
2) Execute `SELECT test_refcursor('my_ref');` in DBeaver 6.1.0
3) Review logs in postgres again and see the **unexpected**:
```
2019-06-13 22:04:51.506 CDT [19591] LOG:  execute <unnamed>: SELECT test_refcursor('my_ref')
2019-06-13 22:04:51.510 CDT [19591] LOG:  execute <unnamed>: FETCH ALL IN "my_ref"
2019-06-13 22:04:51.511 CDT [19591] LOG:  execute <unnamed>: CLOSE "my_ref"
```
3a) All within a matter of milliseconds, dbeaver asked the postgresql server to call the funciton, then interestingly did a `"FETCH ALL IN "my_ref"` on its own, and also a `"CLOSE "my_ref"` Therefore, at this point the `my_ref` cursor is no longer valid.
3b) The dbeaver UI returns `-- refcursor` instead of `my_ref` presumably because it already closed the cursor.
4) Execute `FETCH ALL IN "my_ref";` in DBeaver 6.1.0
5) Review logs in postgres again and see the **unexpected**:
```
2019-06-13 22:11:26.026 CDT [19591] LOG:  execute <unnamed>: FETCH ALL IN "my_ref"
2019-06-13 22:11:26.026 CDT [19591] ERROR:  cursor "my_ref" does not exist
2019-06-13 22:11:26.026 CDT [19591] STATEMENT:  FETCH ALL IN "my_ref"
```
5a) Of course this is unexpected from the end-user experience, but explains why I see the error in dbeaver that `my_ref` does not exist - it's because dbeaver already unexpectedly closed the cursor in step 3.

**Comment by serge-rider** [comment]
Thanks for details, now I see it.
Cursor name extraction will be fixed in 6.1.1. You can test the fix in Early Access version (https://dbeaver.io/files/ea)

Note: Since 6.0.5 you can see actual cursor value by pressing <kbd>shift+enter</kbd> on cursor cell value. So you don't need to write query `FETCH ALL` manually to view the value.

**Comment by uslss** [comment]
verified
